src/styles/blocks.scss

```diff
@@ -0,0 +1,16 @@
+.calendar__block {
+  height: $block-height;
+  width: $block-width;
+  background-color: #eee;
+  border: 1px solid black;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  transition: transform 0.5s;
```

Comment:
Its working only for transition. You need to smoothly change color too. For that you need to use transition-duration: 0.5s;
```suggestion
  transition-duration: 0.5s;
```